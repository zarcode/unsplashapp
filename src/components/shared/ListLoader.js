import React from 'react';
import { View, ActivityIndicator, StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  spinnerWrap: {
    height: 50,
    paddingVertical: 15,
  },
});

const ListLoader = ({ loading }: boolean) => (
  <View style={styles.spinnerWrap}>{loading && <ActivityIndicator />}</View>
);

export default ListLoader;
