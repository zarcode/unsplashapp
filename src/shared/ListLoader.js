import React from 'react';
import { View, ActivityIndicator, StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  spinnerWrap: {
    paddingVertical: 15,
  },
})

const ListLoader = () => (
  <View style={styles.spinnerWrap}>
    <ActivityIndicator />
  </View>
);

export default ListLoader;
