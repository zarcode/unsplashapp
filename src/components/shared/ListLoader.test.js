import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import ListLoader from './ListLoader';

describe('<ListLoader />', () => {
  const wrapper = shallow(<ListLoader />);
  it('renders correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
