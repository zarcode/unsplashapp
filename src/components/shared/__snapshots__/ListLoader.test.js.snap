// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListLoader /> renders correctly 1`] = `
<Component
  style={
    Object {
      "paddingVertical": 15,
    }
  }
>
  <Component />
</Component>
`;
